contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,271 @@
-  std::function<void(Status status)> on_done_;
-};
+              if (status.ok()) {
+                pChecker->env_->LogDebug(
+                    std::string("Test API succeeded with ") + body);
+                status = pChecker->ParseTestResponse(context, &body);
+              } else {
+                pChecker->env_->LogError(std::string("Test API failed with ")
+                    + status.ToString());
+                status = Status(Code::INTERNAL, kFailedFirebaseTest);
+              }
 
-AuthzChecker::AuthzChecker(std::shared_ptr<context::RequestContext> context,
-                           std::function<void(Status status)> continuation)
-    : context_(context),
-      env_(context_->service_context()->env()),
-      on_done_(continuation) {}
+              continuation(status);
+            });
+}
 
-void AuthzChecker::Check() {
-  // TODO: Check service config to see if "useSecurityRules" is specified.
-  // If so, call Firebase Rules service TestRuleset API.
+std::pair<Status, std::string> AuthzChecker::ParseReleaseResponse(
+    std::string *json_str) {
+  grpc_json *json = grpc_json_parse_string_with_len(
+      const_cast<char *>(json_str->data()), json_str->length());
+
+  if (!json) {
+    return std::make_pair(Status(Code::INVALID_ARGUMENT, kInvalidResponse),
+                            "");
+  }
+
+  const char *ruleset_id = GetStringValue(json, "rulesetName");
+  std::pair<Status, std::string> result = std::make_pair(Status::OK,
+                                                         ruleset_id);
+  if (ruleset_id == nullptr) {
+    env_->LogError("Empty ruleset Id received from firebase service");
+    result =std::make_pair(Status(Code::INTERNAL, kInvalidResponse), "");
+  } else {
+    env_->LogInfo(std::string("Received ruleset Id: ") + ruleset_id);
+  }
+
+  grpc_json_destroy(json);
+  return result;
+}
+
+Status AuthzChecker::ParseTestResponse(
+    std::shared_ptr<context::RequestContext> context,
+    std::string *json_str) {
+  grpc_json *json = grpc_json_parse_string_with_len(
+      const_cast<char *>(json_str->data()), json_str->length());
+
+
+  if (!json) {
+    return Status(Code::INVALID_ARGUMENT,
+                  "Invalid JSON response from Firebase Service");
+  }
+
+  Status status = Status::OK;
+  Status invalid = Status(Code::INTERNAL, kInvalidResponse);
+
+  const grpc_json *testResults = GetProperty(json, "testResults");
+  if (testResults == nullptr) {
+    env_->LogError("TestResults are null");
+    status = invalid;
+  } else {
+    const char *result = GetStringValue(testResults->child, "state");
+    if (result == nullptr) {
+      env_->LogInfo("Result state is empty");
+      status = invalid;
+    } else if (std::string(result) != "SUCCESS") {
+      status = Status(Code::PERMISSION_DENIED,
+              std::string("Unauthorized ")
+              + context->request()->GetRequestHTTPMethod()
+              + " access to resource " + context->request()->GetRequestPath(),
```

Comment:
No. Unfortunatley, the Test API that we use only provided with a response that looks like this: 

{
  "testResults": [
    {
      "state": "SUCCESS"
    }
  ]
}

or 

{
  "testResults": [
    {
      "state": "FAILURE"
    }
  ]
}

The response itself is a HTTP 200 OK message.